src/api/users/tools/migrate/migrate.js

```diff
@@ -0,0 +1,68 @@
+const fetch = require('node-fetch');
+const jsdom = require('jsdom');
+const { isWebUri } = require('valid-url');
+const fs = require('fs');
+const { logger } = require('@senecacdot/satellite');
+
+const { JSDOM } = jsdom;
+const { FEED_URL } = process.env;
+
+const FILE = 'legacy_users.json';
+
+const getWikiText = async (url) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.text();
+
+    const dom = new JSDOM(data);
+    return dom.window.document.querySelector('pre').textContent;
+  } catch {
+    throw new Error(`Unable to download wiki feed data from url ${url}`);
+  }
+};
+
+(async () => {
+  const commentRegex = /^\s*#/;
+  let wikiText;
+  let firstName;
+  let lastName;
+  const users = [];
+
+  // Try to fetch the feed list from 'FEED_URL'
+  try {
+    wikiText = await getWikiText(
+      FEED_URL || 'https://wiki.cdot.senecacollege.ca/wiki/Planet_CDOT_Feed_List'
+    );
+  } catch (error) {
+    logger.error(error);
+    process.exit(1);
+  }
+
+  // store every line in an array
+  const lines = wikiText.split(/\r\n|\r|\n/);
+
+  // Iterate through all lines and find url/name pairs, then parse them.
+  lines.forEach((line, index) => {
+    const feeds = [];
+
+    if (!commentRegex.test(line) && line.startsWith('[')) {
+      feeds.push(line.replace(/[[\]']/g, ''));
+
+      if (feeds.length && isWebUri(feeds[0])) {
+        [firstName, lastName] = lines[index + 1].replace(/^\s*name\s*=\s*/, '').split(' ');
+        users.push({ firstName, lastName, feeds });
+      } else {
+        logger.error(`Skipping invalid wiki feed url ${feeds} for author ${firstName} ${lastName}`);
+      }
+    }
```

Comment:
Should we add a `if` statement here to check `wikiText`? The following lines are all dependant on the whether we can get the feed data from the URL. We should also move `const users = [];` down here if we are checking `wikiText` since we don't use the variable yet and can avoid declaring it if `await getWikiText()` fails.

Reply:
I derped out and can't read code. Just noticed the `process.exit(1)`

Reply:
Fixed in [`3c5aba4d`](https://github.com/Seneca-CDOT/telescope/pull/2096/commits/3c5aba4dca6b11754c1932129ebdc0476d822b2e), let me know what you think!